### PR TITLE
openvino 2024.1.0

### PR DESCRIPTION
**Pull request by artanokhov**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by artanokhov** [comment]
@chenrui333 @ZhongRuoyu 
Currently we observe
`Truncated linkage --test openvino output
[46](https://github.com/Homebrew/homebrew-core/actions/runs/8798103438/job/24144603076?pr=169752#step:4:48)  Error: Missing libraries:
[47](https://github.com/Homebrew/homebrew-core/actions/runs/8798103438/job/24144603076?pr=169752#step:4:49)    libze_loader.so.1` 
but we expect libze_loader.so to be part of the driver that the end user can install on their hardware.
Is there a way to suppress this check for a specific library?

**Comment by ZhongRuoyu** [comment]
> Is there a way to suppress this check for a specific library?

No, I don't think so; cc @Homebrew/core

**Comment by SMillerDev** [comment]
> but we expect libze_loader.so to be part of the driver that the end user can install on their hardware.
Is there a way to suppress this check for a specific library?

That error suggests that there is a link to it though, so the user MUST install the driver otherwise it does not work.

**Comment by artanokhov** [comment]
> > but we expect libze_loader.so to be part of the driver that the end user can install on their hardware.
> > Is there a way to suppress this check for a specific library?
> 
> That error suggests that there is a link to it though, so the user MUST install the driver otherwise it does not work.

Yes, that's right, but that's one of the plugins we have. 
Thus, the user does not necessarily need to install the NPU driver if, for example, he will only use the CPU plugin.

So, in other words there is no way to suppress such error?
Because in Conda(Conda-Forge) there is possibility to do that via `missing_dso_whitelist`
 https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#whitelisting-shared-libraries

**Comment by ZhongRuoyu** [comment]
Is `libze_loader.so` from https://github.com/oneapi-src/level-zero? If so, does it make sense to make a formula for https://github.com/oneapi-src/level-zero and add a dependency on it?

File: Formula/o/openvino.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Openvino < Formula
   desc "Open Visual Inference And Optimization toolkit for AI inference"
   homepage "https://docs.openvino.ai"
-  url "https://github.com/openvinotoolkit/openvino/archive/refs/tags/2024.0.0.tar.gz"
-  sha256 "b3c257f8af9545ae68a6ea217173b2b2de9dd42d35e8703a7a51d76f4c2bfe2f"
+  url "https://github.com/openvinotoolkit/openvino/archive/refs/tags/2024.1.0.tar.gz"
+  sha256 "b298a91b5aae252ef9883e0f2017e88677be88a9839b1aa2f6e9f70067d98ce6"
   license "Apache-2.0"
   head "https://github.com/openvinotoolkit/openvino.git", branch: "master"
```
Comment (ilya-lavrenov) [inline Formula/o/openvino.rb:8]: BTW, let's drop `cython`. It's not used since 2024.0

**Comment by ilya-lavrenov** [comment]
@ZhongRuoyu @chenrui333
is it possible to implement auto-bump of dependencies' revisions? It can be analyzed that one package hardly depends on another and perform auto-bump

**Comment by artanokhov** [comment]
@ZhongRuoyu  @chenrui333 
As I see currently CI has fallen due to
`Error: PR requires the CI-long-timeout label but it is not set!
Error: If the longer timeout is not required, remove the "long build" label.
Error: Otherwise, add the "CI-long-timeout" label.
Error: No more than 2 PRs at a time may use "CI-long-timeout".`

Please clarify whether it will be set automatically or will some additional actions be required?

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**